lib/IDE/Refactoring.cpp

```diff
@@ -874,10 +896,9 @@ static void analyzeRenameScope(ValueDecl *VD, Optional<RenameRefInfo> RefInfo,
   }
 
   auto *Scope = VD->getDeclContext();
-  // If the context is a top-level code decl, there may be other sibling
-  // decls that the renamed symbol is visible from
-  if (isa<TopLevelCodeDecl>(Scope))
+  while (canDeclContextElementsBeReferencedFromOutside(Scope)) {
```

Comment:
There is a slight change in behavior here which you noticed in the other PR (which doesn't work without this one). This is essentially `getModuleScopeContext`, which means that now rename is no longer "local".

@benlangmuir I'm not sure whether this matters or not. It seems like it'd make sense to just return all the locations we know about within the module, but maybe there was a reason it wasn't doing that?

Depending on that answer I'd either use `getModuleScopeContext` or change to grab the last context *before* the module/file context.